### ISSUE TITLE
Mvvmcross 6.2.3 strange behavior of viewmodel navigation [or bug?]

### ISSUE THREAD

**Issue by JTOne123** [MvvmCross/MvvmCross#3342]
I'm playing with MVVMCROSS 6.2.3 and I have an issue with nested navigation in my XForms app (tested on UWP and Android).

Here is a description: VM1 navigate to VM2 and VM1 is waiting for a result of VM2!
VM2 navigate to VM3, VM2 does not need a result of VM3. After navigation from VM2 to VM3 and VM1 is unexpectedly getting a null result.

```
public class MainViewModel : MvxViewModel
{
    public IMvxAsyncCommand GoToSecondPageCommand =>
        new MvxAsyncCommand(async () =>
        {
            var param = new Dictionary<string, string> { { "ButtonText", ButtonText } };

            var result = await navigationService.Navigate<SecondViewModel, Dictionary<string, string>, SecondViewModelResult>(param);
            var breakpoint = 1;


            //HERE IS THE ISSUE, result IS NULL because it fires immediately after Navigate<ThirdViewModel>()


        });

    public IMvxAsyncCommand BackCommand => new MvxAsyncCommand(async () =>
    {
        await navigationService.Close(this, new SecondViewModelResult { Result = "Back button clicked" });
    });
}

public class SecondViewModel : MvxViewModel<Dictionary<string, string>, SecondViewModelResult>
{
    public IMvxAsyncCommand GoToThirdPageCommand =>
        new MvxAsyncCommand(async () =>
        {
            var t = await navigationService.Navigate<ThirdViewModel>();
            var breakpoint = 2;
        });
}

public class ThirdViewModel : MvxViewModel
{
}
```

You could run it by yourself from my GitHub https://github.com/JTOne123/XamFormsMvxTemplate

This behavior was good in version 5.7 and VM1 is waiting until VM2 return the result.

What am I doing wrong?

**Comment by JTOne123** [comment]
It's easily reproduced in Playground project

Set breakpoint here
![image](https://user-images.githubusercontent.com/3457140/55938745-b3779180-5c44-11e9-9dac-9867c8f33588.png)

1. Go to Show Child
2. Go to Show Second Child
3. Wait for 5 second
4. You will be stopped in **var testIfReturn = result;** breakpoint

I'm digging deeper...

**Comment by JTOne123** [comment]
Current behavior was a consequence of next commit https://github.com/MvvmCross/MvvmCross/commit/32b6868e84740e6b7c35a56822736702989b325e

After I've commented this line and everything works how it expected.
![image](https://user-images.githubusercontent.com/3457140/55950501-c21e7280-5c5d-11e9-9c88-12def5f45ea9.png)

---

That's mean no more exists a sequence of viewmodel's navigations because the page disappearing is calling a destroy of VM (**We should understand that page is still alive and might be shown in future**)

@martijn00 are you pursue this goal and my question is invalid? or we should revert your commit?

**Comment by bravekit** [comment]
I had the same behavior with tabbed pages in my project
I navigate from VM1  to VM2 with a tappedpage view (one VM2 for all tabs). VM1 is waiting for a result from VM2. For the first time when I'm switching from tab1 to any tab all is ok, but the second switch to any tab leads to null result in VM1, although the VM2 and its views are still exist
 Eventually, I decided to use MVVMCROSS 6.1.2 which doesn't have this issue

**Comment by foehammer88** [comment]
I'm still experiencing this issue in 6.3.1.

**Comment by vvche69** [comment]
Any news?
This issue still exists in version 7.1.1.

**Comment by Cheesebaron** [comment]
@vvche69 nope. Comments like yours are not helpful either. If you want this fixed, roll up the sleeves and create a PR to fix this.

**Comment by vvche69** [comment]
Ok, I've opened a new issue.

**Comment by Cheesebaron** [comment]
@vvche69 why? It is a duplicate of this one.

**Comment by Muck92** [comment]
Is there a chance to fix this issue?

**Comment by bravekit** [comment]
I can sugggest a workaround
Your model VM1 navigate to VM2 and VM1 is waiting for a result of VM2. You should pass a TaskCompletionSource to VM2, and wait for its result from VM2

Code for VM1, when it is navigaing to VM2
`var tcs = new TaskCompletionSource<bool>();
            var result = await Navigation.Navigate<VM2, object, object>(tcs);
            result = await tcs.Task.ConfigureAwait(false);`

When VM2 is being closed, it must set result for this task completition sorce, thus this will unlock VM1 for further executing code 
Code for VM2
` private void CloseVM2Model()
        {
            Navigation.Close(this);
            _tcs?.TrySetResult(true);
        }`

**Comment by Muck92** [comment]
@bravekit Thanks, this workaround works for me :)
